src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const properties = sourceString.split(';');
+
+  properties.forEach((element) => {
+    let [property, value] = element.split(':');
+
+    if (property && value) {
+      property = property.trim();
+      value = value.trim();
+
+      result[property] = value;
+    }
+  });
```

Comment:
Instead of using 'forEach' and modifying the 'result' object within it, consider using 'reduce' to build the object in an immutable way. This aligns with the checklist point about avoiding direct modification of an existing object within iteration methods.